### PR TITLE
Added country/region support for locale when displayed for sources

### PR DESCRIPTION
**Pull request by CarlosEsco**
Adjusted locale from external sources and catalogues to use the same code.  Originally was planning to use it an kotlin extension but decided to stuff in in the Locale Helper for consistency sake.

### CODE REVIEW

File: app/src/main/java/eu/kanade/tachiyomi/ui/extension/ExtensionDetailsController.kt [hunk 1]
```diff
@@ -122,8 +123,8 @@ class ExtensionDetailsController(bundle: Bundle? = null) :
         val dataStore = SharedPreferencesDataStore(/*if (source is HttpSource) {
             source.preferences
         } else {*/
-            context.getSharedPreferences("source_${source.id}", Context.MODE_PRIVATE)
-        /*}*/)
+                context.getSharedPreferences("source_${source.id}", Context.MODE_PRIVATE)
+                /*}*/)
```
Comment (paronos) [inline app/src/main/java/eu/kanade/tachiyomi/ui/extension/ExtensionDetailsController.kt]: the additional spaces are desidered?

File: app/src/main/java/eu/kanade/tachiyomi/util/LocaleHelper.kt [hunk 2]
```diff
@@ -45,10 +47,33 @@ object LocaleHelper {
...
-        val country = parts.getOrNull(1) ?: ""
-        return Locale(lang, country)
+        return getLocale(pref)
+    }
+
+    /**
+     * Returns the locale for the value stored in preferences, or null if it's system language.
+     *
+     * @param pref the string value stored in preferences.
+     */
```
Comment (paronos) [inline app/src/main/java/eu/kanade/tachiyomi/util/LocaleHelper.kt]: this description has some problems

File: app/src/main/java/eu/kanade/tachiyomi/util/LocaleHelper.kt [hunk 3]
```diff
@@ -45,10 +47,33 @@ object LocaleHelper {
...
+            "all" -> context.getString(R.string.all_lang)
+            else -> {
+                val locale = getLocale(lang)
+                locale.getDisplayName(locale).capitalize()
+            }
+        }
+    }
+
+    private fun getLocale(lang: String): Locale {
+        val sp = lang!!.split("_", "-")
```
Comment (paronos) [inline app/src/main/java/eu/kanade/tachiyomi/util/LocaleHelper.kt]: Is `!!` needed?

File: app/src/main/java/eu/kanade/tachiyomi/ui/extension/ExtensionDetailsController.kt (hunk 1 above)
Comment (CarlosEsco) [reply app/src/main/java/eu/kanade/tachiyomi/ui/extension/ExtensionDetailsController.kt]: Thats how code was then i used the default kotlin formatting

File: app/src/main/java/eu/kanade/tachiyomi/util/LocaleHelper.kt (hunk 3 above)
Comment (CarlosEsco) [reply app/src/main/java/eu/kanade/tachiyomi/util/LocaleHelper.kt]: fixed

File: app/src/main/java/eu/kanade/tachiyomi/util/LocaleHelper.kt (hunk 2 above)
Comment (CarlosEsco) [reply app/src/main/java/eu/kanade/tachiyomi/util/LocaleHelper.kt]: fixed

**Pull request merged**